aptos-move/framework/aptos-experimental/sources/trading/market/pre_cancellation_tracker.move

```diff
@@ -0,0 +1,227 @@
+module aptos_experimental::pre_cancellation_tracker {
+    use std::signer;
+    use aptos_std::big_ordered_map;
+    use aptos_std::big_ordered_map::BigOrderedMap;
+    use aptos_experimental::order_book_types::{
+        AccountClientOrderId,
+        new_account_client_order_id
+    };
+    #[test_only]
+    use std::vector;
+    #[test_only]
+    use aptos_framework::timestamp;
+
+    const DUPLICATE_ORDER_PLACEMENT: u64 = 1;
+
+    const MAX_ORDERS_GARBAGE_COLLECTED_PER_CALL: u64 = 10;
+
+    struct PreCancellationTracker has store {
+        pre_cancellation_window_secs: u64,
+        // Map of order IDs with expiration time to a boolean indicating if the order is active.
+        expiration_with_order_ids: BigOrderedMap<ExpirationAndOrderId, bool>,
+        // Map of order Ids to their corresponding expiration time.
+        account_order_ids: BigOrderedMap<AccountClientOrderId, u64>
+    }
+
+    struct ExpirationAndOrderId has copy, drop, store {
+        expiration_time: u64,
+        account_order_id: AccountClientOrderId
+    }
+
+    public fun new_pre_cancellation_tracker(expiration_time_secs: u64): PreCancellationTracker {
```

Comment:
Make all functions here to be public(package) for easier compatibility

Reply:
Yes,  anything not required externally should be friend visibility (we don't use package in framework,  as it's too large)